src/utils/daos_hdlr.c

```diff
@@ -730,118 +762,175 @@ cont_list_attrs_hdlr(struct cmd_args_s *ap)
 
 }
 
-/* cont_get_prop_hdlr() - get container properties */
-int
-cont_get_prop_hdlr(struct cmd_args_s *ap)
+static int
+cont_decode_props(daos_prop_t *props)
 {
-	daos_prop_t		*prop_query;
-	struct daos_prop_entry	*entry;
-	char			type[10] = {};
-	int			rc = 0;
-	uint32_t		i;
-	uint32_t		entry_type;
 
-	/*
-	 * Get all props except the ACL
-	 */
-	prop_query = daos_prop_alloc(DAOS_PROP_CO_NUM - 1);
-	if (prop_query == NULL)
-		return -DER_NOMEM;
+	struct daos_prop_entry          *entry;
+	char				type[10];
+	int 				rc = 0;
 
-	entry_type = DAOS_PROP_CO_MIN + 1;
-	for (i = 0; i < prop_query->dpp_nr; entry_type++) {
-		if (entry_type == DAOS_PROP_CO_ACL)
-			continue; /* skip ACL */
-		prop_query->dpp_entries[i].dpe_type = entry_type;
-		i++;
-	}
+	/* unset properties should get default value */
 
-	rc = daos_cont_query(ap->cont, NULL, prop_query, NULL);
-	if (rc) {
-		fprintf(stderr, "Container query failed, result: %d\n", rc);
-		D_GOTO(err_out, rc);
-	}
-
-	D_PRINT("Container properties :\n");
-
-	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LABEL);
+	entry = daos_prop_entry_get(props, DAOS_PROP_CO_LABEL);
 	if (entry == NULL || entry->dpe_str == NULL) {
 		fprintf(stderr, "label property not found\n");
 		D_GOTO(err_out, rc = -DER_INVAL);
 	}
-	D_PRINT("label -> %s\n", entry->dpe_str);
+	D_PRINT("label\t\t\t%s\n", entry->dpe_str);
```

Comment:
For container props analogous to the pool props, for consistent formatting, you could add even smaller helper functions called in both cases, e.g.
`int print_label(struct daos_prop_entry *label_entry)`

Reply:
Well, this concerns only label, owner and owner_group which do not need special handling as they are strings. But to find this I had to review all pool/cont props, and this has permitted me to find that I forgot cont owner/owner_group props display !!